RoboJackets/robocup-software
# Various planning fixes

## Description
Various tweaks and fixes to motion planning. Tuning the dynamic obstacle radius currently.

 * when goal pt is blocked by an obstacle, rather than not planning anything, pick the closest point on the start-end line and use that as the new goal
 * inflate obstacle radius when obstacles have velocity
 * shift opp robot obstacle in direction of travel slightly

per @p-nayak11 comment below, new obstacle shift:
![Figure_1](https://user-images.githubusercontent.com/30275369/174866017-a6e04023-9f59-48ca-9441-798f1bd7fa82.png)

 * other comments/TODOs/style fixes

## Steps to test

### Test Case 1
 * Run the motion planning test play at the same time (from #1906) with sim2play. You should see the robots don't move when their goal points are blocked.
 * Run the same plays on my branch. You should see they move close to the point, meaning they can correct more easily.
 
### Test Case 2
Run various other plays on my branch.

Expected result: soccer does not crash, no obviously worse motion planning than before.
